### PR TITLE
XCode preparing for AppStore

### PR DESCRIPTION
**Pull request by dimitre**
- [x] updating info.plist
- [x] adding of.entitlements
- [x] add of.entitlements to project
- [x] 1024x1024 icons (debug and release) - Those are provisory simplified icons, with only 512 / 1024 dimensions
- [x] create another PR on project generator to add AppStore scheme, and copy over of.entitlements  - https://github.com/openframeworks/projectGenerator/pull/292
- [ ] PG: substitute spaces by underline in scheme names, like "emptyExample Debug" - not to be done now because it can change other things in scripts.

### CODE REVIEW

**Comment by ofTheo** [comment]
@dimitre I have some notes/questions inline above

**Comment by dimitre** [comment]
hey @ofTheo where can I see your notes/questions?

**Review by ofTheo — CHANGES_REQUESTED** [review]

**Comment by ofTheo** [comment]
sorry @dimitre - I guess I started a review weeks ago with comments, but GH doesn't share any of my notes until I submitted the review. I assumed you were seeing all the incline comments as I added them. 🤦‍♂️

**Comment by dimitre** [comment]
Yeahh now I can see everything> i'll be addressing each one tomorrow :)

**Comment by ofTheo** [comment]
Thanks! And apologies!! 🙂👍

File: scripts/templates/osx/of.entitlements [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>com.apple.security.app-sandbox</key>
+	<true/>
```
[the remark this answers is not in the document]
Comment (dimitre) [reply scripts/templates/osx/of.entitlements:22]: Thanks @ofTheo, good catch. I think this particular review was from an earlier commit, this is already fixed

File: libs/openFrameworksCompiled/project/osx/CoreOF.xcconfig [hunk 2]
```diff
@@ -53,6 +53,58 @@ OF_CORE_HEADERS = $(HEADER_OF) $(HEADER_FREETYPE) $(HEADER_FREETYPE2) $(HEADER_F
...
+GCC_ENABLE_SSE3_EXTENSIONS = YES
+GCC_ENABLE_SUPPLEMENTAL_SSE3_INSTRUCTIONS = YES
+GCC_INLINES_ARE_PRIVATE_EXTERN = NO
+GCC_SYMBOLS_PRIVATE_EXTERN = NO
+ONLY_ACTIVE_ARCH = YES
+OTHER_CODE_SIGN_FLAGS = --deep
+GCC_GENERATE_DEBUGGING_SYMBOLS = YES
+GCC_MODEL_TUNING = NONE
+COMBINE_HIDPI_IMAGES = YES
+INFOPLIST_FILE = openFrameworks-Info.plist
```
[the remark this answers is not in the document]
Comment (dimitre) [reply libs/openFrameworksCompiled/project/osx/CoreOF.xcconfig]: Great idea, just moved this setting to the Project.xcconfig

File: scripts/templates/osx/Project.xcconfig [hunk 3]
```diff
@@ -9,73 +9,34 @@ OF_PATH = ../../..
...
 //OF_CORE_HEADERS = $(inherited) $(HEADER_BOOST)
 
-//ICONS - NEW IN 0072 
-ICON_NAME = icon.icns
-ICON_NAME[config=Debug] = icon-debug.icns
-ICON_FILE = $(OF_PATH)/libs/openFrameworksCompiled/project/osx/$(ICON_NAME)
-
-//IF YOU WANT AN APP TO HAVE A CUSTOM ICON - PUT THEM IN YOUR DATA FOLDER AND CHANGE ICON_FILE_PATH to:
-//ICON_FILE_PATH = bin/data/
+#include? "App.xcconfig"
```
[the remark this answers is not in the document]
Comment (dimitre) [reply scripts/templates/osx/Project.xcconfig:40]: Thanks Theo, I've just added a comment, let me know if it is ok.
I still remember our conversation and I'll be happy to totally remove it in near future when we stabilize all settings before next release, if needed. :)

**Review by ofTheo — APPROVED** [review]

**Pull request merged**